src/Dependencies/Collections/ImmutableSegmentedDictionary`2.cs

```diff
@@ -70,7 +70,18 @@ namespace Microsoft.CodeAnalysis.Collections
     internal readonly partial struct ImmutableSegmentedDictionary<TKey, TValue> : IImmutableDictionary<TKey, TValue>, IDictionary<TKey, TValue>, IReadOnlyDictionary<TKey, TValue>, IDictionary, IEquatable<ImmutableSegmentedDictionary<TKey, TValue>>
         where TKey : notnull
     {
-        public static readonly ImmutableSegmentedDictionary<TKey, TValue> Empty = new(new SegmentedDictionary<TKey, TValue>());
+        public static readonly ImmutableSegmentedDictionary<TKey, TValue> Empty = CreateEmpty();
+
+        /// <summary>
+        /// Creates a new empty segmented dictionary, that is not underlying-reference equal to <see cref="Empty"/>. The intent is for this to
+        /// be used for sentinel values only.
+        /// </summary>
+        internal static ImmutableSegmentedDictionary<TKey, TValue> CreateEmpty() => new(new SegmentedDictionary<TKey, TValue>());
```

Comment:
📝 This doesn't seem like a generally-valuable API. For the edge case where it's needed, it would be better to construct the sentinel instance using private reflection rather than facilitate this type of usage in other scenarios.